core/controllers/library_test.py

```diff
@@ -194,14 +197,148 @@ def test_library_handler_for_created_explorations(self):
         }, response_dict['activity_list'][0])
 
 
+class LibraryIndexHandlerTests(test_utils.GenericTestBase):
+
+    def setUp(self):
+        super(LibraryIndexHandlerTests, self).setUp()
+        self.signup(self.VIEWER_EMAIL, self.VIEWER_USERNAME)
+        self.viewer_id = self.get_user_id_from_email(self.VIEWER_EMAIL)
+
+    def test_library_index_handler_for_user_preferred_language(self):
+        """Test whether the handler returns the correct language preference."""
+        # Since the default language is 'en', the language preference for the
+        # viewer is changed to 'de' to test if the preference returned is user's
+        # preference.
+        self.login(self.VIEWER_EMAIL)
+        response = self.get_html_response('/preferences')
+        csrf_token = self.get_csrf_token_from_response(response)
+
+        # Change the user's preferred language to de.
+        self.put_json(
+            '/preferenceshandler/data',
+            {'update_type': 'preferred_language_codes', 'data': ['de']},
+            csrf_token=csrf_token)
+        response_dict = self.get_json(feconf.LIBRARY_INDEX_DATA_URL)
+        self.assertDictContainsSubset({
+            'activity_summary_dicts_by_category': [],
+            'preferred_language_codes': ['de'],
+        }, response_dict)
+
+    def test_library_index_handler_for_top_rated(self):
```

Comment:
Maybe name should be something like `test_library_index_handler_updates_top_rated_activity_summary_dicts`

Reply:
renamed